cli/src/main/scala/com/codacy/analysis/cli/configuration/CLIProperties.scala

```diff
@@ -0,0 +1,199 @@
+package com.codacy.analysis.cli.configuration
+
+import better.files.File
+import cats.Foldable
+import cats.implicits._
+import com.codacy.analysis.cli.command.Analyse
+import com.codacy.analysis.cli.configuration.CLIProperties.{AnalysisProperties, ResultProperties, UploadProperties}
+import com.codacy.analysis.core.clients.CodacyClient
+import com.codacy.analysis.core.clients.api._
+import com.codacy.analysis.core.configuration.{CodacyConfigurationFile, EngineConfiguration}
+import com.codacy.analysis.core.files.Glob
+import com.codacy.analysis.core.git.{Commit, Git}
+import com.codacy.analysis.core.utils.MapOps
+import com.codacy.plugins.api.languages.Language
+import play.api.libs.json.JsValue
+
+import scala.concurrent.duration.Duration
+
+case class CLIProperties(analysis: AnalysisProperties, upload: UploadProperties, result: ResultProperties)
+
+object CLIProperties {
+
+  // HACK: Fixes Intellij IDEA highlight problems
+  private type EitherA[A] = Either[String, A]
+  private val foldable: Foldable[EitherA] = implicitly[Foldable[EitherA]]
+
+  case class AnalysisProperties(projectDirectory: File,
+                                output: AnalysisProperties.Output,
+                                tool: Option[String],
+                                parallel: Option[Int],
+                                forceFilePermissions: Boolean,
+                                fileExclusionRules: AnalysisProperties.FileExclusionRules,
+                                toolProperties: AnalysisProperties.Tool)
+
+  object AnalysisProperties {
+
+    case class Tool(toolTimeout: Option[Duration],
```

Comment:
I agree with you for most case classes.
The next commit will contain changes to reflect this with the exception of the case classes that are used for the definition of CLIConfiguration.IssuesTool, since these ones need the extra name in its namespace to make proper sense (like CLIConfiguration.IssuesTool.Parameter vs CLIConfiguration.Parameter).